tests/unit/models/test_base_model.py

```diff
@@ -388,6 +388,65 @@ def test_from_dict_submit(self):
         actual = Request.from_dict(request)
         self.assertEqual(actual, expected)
 
+    # Note: BaseModel.from_xrpl and its overridden methods accept only camelCase or
+    # PascalCase inputs (i.e. snake_case is not accepted)
+    def test_from_xrpl_accepts_only_camel_case_inputs(self):
+        blob = "SOISUSF9SD0839W8U98J98SF"
+        id_val = "submit_786514"
+        request = {
+            "method_name": "submit",
+            "tx_blob": blob,
+            "fail_hard": False,
+            "id_in_snake_case": id_val,
+        }
+
+        with self.assertRaises(XRPLModelException):
+            Request.from_xrpl(request)
```

Comment:
I don't think this is a good test case. This is how requests do actually look, it's just not how you're supposed to generate requests.

Reply:
okay, I have modified the request to use a tx_json. please check [244369c](https://github.com/XRPLF/xrpl-py/pull/694/commits/244369cdc80a959c28f70e23ce7d9ba7df07c890)